### PR TITLE
Update PowerShell for .NET 6, .NET 7 and .NET 8

### PR DESCRIPTION
**Pull request by adityapatwardhan**
Updating Powershell for .NET 6, .NET 7 and .NET 8 over main branch.

### CODE REVIEW

File: manifest.versions.json [hunk 1]
```diff
@@ -49,15 +49,15 @@
     "base-url|6.0|nightly": "$(base-url|public|maintenance|nightly)",
     "base-url|7.0|main": "$(base-url|public|maintenance|main)",
     "base-url|7.0|nightly": "$(base-url|public|maintenance|nightly)",
-    "base-url|8.0|main": "$(base-url|public|main)",
-    "base-url|8.0|nightly": "$(base-url|public|nightly)",
+    "base-url|8.0|main": "$(base-url|public|maintenance|main)",
```
Comment (mthalman) [inline manifest.versions.json:52]: Why are these maintenance changes needed?

File: manifest.versions.json [hunk 2]
```diff
@@ -182,7 +182,7 @@
     "runtime|8.0|win|x64|sha": "b786869739c87ea286ee494fbf72cc92cf0cd91c1f22ede0133a523aa150740dd89cdca6a1530bd1f39d24db4f34939e17e2b0dce3d7efd06e4eced30b530643",
 
     "sdk|6.0|build-version": "6.0.417",
-    "sdk|6.0|product-version": "6.0.417",
+    "sdk|6.0|product-version": "6.0.417-1",
```
[the remark this answers is not in the document]
Comment (lbussell) [reply manifest.versions.json:185]: The tags look correct to me - https://dev.azure.com/dnceng-public/public/_build/results?buildId=473087&view=logs&j=739bb7ec-7296-5b40-7dff-ee3297bfdaa4&t=729bd3d3-63cd-5b66-6efd-ae98d0926964&l=336 [internal link]

`product-version` affects the tags, and `build-version` is the only thing that makes it into the Dockerfiles.

File: manifest.versions.json (hunk 1 above)
Comment (lbussell) [reply manifest.versions.json:52]: Not sure how that made it in there, must have been left over from a previous checkout. Will revert that.

File: eng/dockerfile-templates/sdk/Dockerfile.linux.install-powershell [hunk 3]
```diff
@@ -4,7 +4,7 @@
     set nupkgPlatform to when(isAlpine, "Linux.Alpine", cat("Linux.", ARCH_NUPKG)) ^
     set shaPlatform to when(isAlpine, "Linux.Alpine", cat("Linux|", ARCH_NUPKG)) ^
     set nupkgName to cat("PowerShell.", nupkgPlatform, ".$powershell_version.nupkg") ^
-    set url to cat("https://pwshtool.blob.core.windows.net/tool/$powershell_version/", nupkgName)
+    set url to cat("https://powershellinfraartifacts-gkhedzdeaghdezhr.z01.azurefd.net/tool/$powershell_version/", nupkgName)
```
Comment (mthalman) [inline eng/dockerfile-templates/sdk/Dockerfile.linux.install-powershell:7]: @adityapatwardhan - Is this a permanent change or just something specific to this release?

**Review by mthalman — APPROVED** [review]

**Pull request merged**